src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs

```diff
@@ -80,6 +83,17 @@ public MessageMetadata GetMessageMetadata(string messageTypeIdentifier)
             return null;
         }
 
+        string GetMessageTypeNameWithoutAssembly(string messageTypeIdentifier)
+        {
+            var typeParts = messageTypeIdentifier.Split(',');
+            if (typeParts.Length > 1)
+            {
+                return typeParts[0]; //Take the type part only
+            }
+
+            return messageTypeIdentifier;
```

Comment:
Consider
`return Type.GetType(messageTypeIdentifier)?.FullName ?? messageTypeIdentifier;`

Reply:
@deeess239 that wouldn't work. The main issue here is that the Type doesn't exist (or rather exist in an assembly that is not loaded) so when we do `Type.GetType` it will return `NULL`.